openmc/kinetics/solver.py

```diff
@@ -0,0 +1,831 @@
+        f.attrs['use_pcmfd'] = self.use_pcmfd
+        f.attrs['num_delayed_groups'] = self.num_delayed_groups
+        f.attrs['num_outer_time_steps'] = self.num_outer_time_steps
+        f.attrs['use_pregenerated_sps'] = self.use_pregenerated_sps
+        f.attrs['constant_seed'] = self.constant_seed
+        f.attrs['seed'] = self.seed
+        f.attrs['core_volume'] = self.core_volume
+        f.attrs['k_crit'] = self.k_crit
+        f.attrs['method'] = self.method
+        f.attrs['pnl'] = self.states['START'].pnl
+        f.require_group('clock')
+        f['clock'].attrs['dt_outer'] = self.clock.dt_outer
+        f['clock'].attrs['dt_inner'] = self.clock.dt_inner
+        f.require_group('OUTER_STEPS')
+        f.require_group('INNER_STEPS')
+        f.create_dataset('beta', data=self.states['START'].beta())
+        f.create_dataset('beta_shape', data=self.states['START'].beta_shape())
+        f.create_dataset('beta_eff', data=self.states['START'].beta_eff)
+        f.create_dataset('power', data=self.states['START'].power)
+        f.create_dataset('decay_rate', data=self.states['START'].decay_rate)
+        f.create_dataset('inverse_velocity', data=self.states['START'].inverse_velocity)
+        total = self.states['START'].absorption + self.states['START'].outscatter
+        f.create_dataset('total', data=total)
+
+        f.close()
+
+    def run_openmc(self, time_point, method='ADIABATIC'):
+
+        self.setup_openmc(time_point, method)
+
+        # Names of the statepoint and summary files
+        sp_old_name = '{}/statepoint.{}.h5'.format(self.directory, self.settings_file.batches)
+        sp_new_name = '{}/statepoint_{:.6f}_sec.{}.h5'\
+                      .format(self.directory, self.clock.times[time_point], self.settings_file.batches)
+        sum_old_name = '{}/summary.h5'.format(self.directory)
+        sum_new_name = '{}/summary_{:.6f}_sec.h5'.format(self.directory, self.clock.times[time_point])
+        old_source = '{}/source.{}.h5'.format(self.directory, self.settings_file.batches)
+        new_source = '{}/source.h5'.format(self.directory)
+
+        # Run OpenMC
+        if not self.use_pregenerated_sps:
+            openmc.run(threads=self.threads, mpi_args=self.mpi_args,
+                           cwd=self.directory)
+
+            # Rename the statepoint and summary files
+            copyfile(sp_old_name, sp_new_name)
+            copyfile(sum_old_name, sum_new_name)
+            move(old_source, new_source)
+
+        # Load the summary and statepoint files
+        summary_file = openmc.Summary(sum_new_name)
+        statepoint_file = openmc.StatePoint(sp_new_name, False)
+        statepoint_file.link_with_summary(summary_file)
+
+        # Load mgxs library
+        for mgxs in self.states[time_point].mgxs_lib.values():
+            mgxs.load_from_statepoint(statepoint_file)
+
+        self.states[time_point].load_mgxs()
+
+    def create_state(self, time_point):
```

Comment:
If this method is not meant to be called by a user (I assume it is not), the convention is to make it private by prepending an underscore, i.e. `def _create_state`